addons/crm/models/crm_lead.py

```diff
@@ -290,54 +270,40 @@ def _onchange_user_values(self, user_id):
         team_id = self._default_team_id(user_id)
         return {'team_id': team_id}
 
-    @api.onchange('user_id')
-    def _onchange_user_id(self):
+    @api.depends('user_id')
+    def _compute_user_values(self):
         """ When changing the user, also set a team_id or restrict team id to the ones user_id is member of. """
-        if self.user_id.sale_team_id:
-            values = self._onchange_user_values(self.user_id.id)
-            self.update(values)
-
-    def _onchange_compute_probability(self, optional_field_name=None):
-        """Recompute probability on onchange methods of :
-            'stage_id', 'team_id', 'tag_ids'
-            'country_id', 'state_id', 'phone_state', 'email_state', 'source_id' """
-        if optional_field_name and optional_field_name not in self._pls_get_safe_fields():
-            return
-        lead_probabilities = self._pls_get_naive_bayes_probabilities()
-        if self.id in lead_probabilities:
-            self.automated_probability = lead_probabilities[self.id]
-            if self._origin.is_automated_probability:
-                self.probability = self.automated_probability
-
-    @api.onchange('stage_id')
-    def _onchange_stage_id(self):
-        self._onchange_compute_probability()
-
-    @api.onchange('source_id')
-    def _onchange_source_id(self):
-        self._onchange_compute_probability(optional_field_name='source_id')
-
-    @api.onchange('automated_probability')
-    def _onchange_automated_probability(self):
-        """ If was in auto mode before the change, align both fields so that it will stay in auto mode. """
-        if self._origin.is_automated_probability:
-            self.probability = self.automated_probability
-
-    @api.onchange('team_id')
-    def _onchange_team_id(self):
-        self._onchange_compute_probability()
-
-    @api.onchange('tag_ids')
-    def _onchange_tag_ids(self):
-        self._onchange_compute_probability()
+        for lead in self:
+            lead.update(lead._onchange_user_values(lead.user_id.id))
```

Comment:
and after refactoring `_onchange_user_values` into `_get_user_team_id`, you can directly do
`lead.team_id = lead._get_user_team_id(lead.user_id.id)`